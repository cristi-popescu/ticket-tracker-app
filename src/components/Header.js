import React from "react";
import { BrowserRouter as Router, Link } from "react-router-dom";
import { connect } from "react-redux";
import Navbar from "react-bootstrap/Navbar";
import Button from "react-bootstrap/Button";
import Logo from "../logo.svg";

import { showTicketModal } from "../actions/ticketActions";

export const Header = props => {
    const handleClick = () => {
        const { showTicketModal } = props;

        showTicketModal({});
    };

    return (
        <Router>
            <Navbar bg="dark" variant="dark">
                <Link to="/">
                    <Navbar.Brand>
                        <img
                            src={Logo}
                            width="50"
                            height="50"
                            className="d-inline-block align-middle"
                            alt="Ticket Tracker"
                        />
                        Ticket Tracker
                    </Navbar.Brand>
                </Link>

                <Link to="/" className="navbar-link">
                    Dashboard
                </Link>

                <Link to="/ticket-list" className="navbar-link">
                    View All Tickets
                </Link>

                <Button variant="light" onClick={handleClick}>
                    Create Ticket
                </Button>
            </Navbar>
        </Router>
    );
};

export default connect(
    null,
    { showTicketModal }
)(Header);
