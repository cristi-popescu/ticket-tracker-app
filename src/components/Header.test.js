import React from "react";
import { Header } from "./Header";
import renderer from "react-test-renderer";

test("Header renders correctly", () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
