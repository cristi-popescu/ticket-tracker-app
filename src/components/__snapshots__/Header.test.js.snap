// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <span
      className="navbar-brand"
    >
      <img
        alt="Ticket Tracker"
        className="d-inline-block align-middle"
        height="50"
        src="logo.svg"
        width="50"
      />
      Ticket Tracker
    </span>
  </a>
  <a
    className="navbar-link"
    href="/"
    onClick={[Function]}
  >
    Dashboard
  </a>
  <a
    className="navbar-link"
    href="/ticket-list"
    onClick={[Function]}
  >
    View All Tickets
  </a>
  <button
    className="btn btn-light"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Create Ticket
  </button>
</nav>
`;
